sdks/java/core/src/test/java/org/apache/beam/sdk/io/CompressedSourceTest.java

```diff
@@ -632,6 +646,27 @@ protected boolean readNextRecord() throws IOException {
       protected long getCurrentOffset() {
         return offset;
       }
+
+      @Override
+      public Instant getCurrentTimestamp() throws NoSuchElementException {
+        Instant timestamp = ((ByteSource) getCurrentSource()).timestamp;
```

Comment:
how about we just return `Duration.millis(getCurrentOffset())`? That way elements are timestamped by their offsets fairly directly.